lib/src/noop_scope.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:opentracing/noop_tracer.dart';
+import 'package:opentracing/opentracing.dart';
+
+/// The No-op implementation of [Scope] in which all operations are no-op
+class NoopScope implements Scope {
+  /// Returns a new NoopScope, which will use the supplied Span. If no Span is
+  /// supplied, a static instance of a NoopSpan will be used.
+  NoopScope({Span span}) {
+    _span = span ?? _span;
```

Comment:
Could overwriting the static _span when a new NoopScope is created have the potential to be problematic? 

For instance:
```dart
final span1 = globalTracer().startSpan('span1');
final span2 = globalTracer().startSpan('span2');

final noopScope1 = noopScopeManager.activate(span1, false);
print(noopScope1.span.operationName); // span1

final noopScope2 = noopScopeManager.activate(span2, false);
print(noopScope1.span.operationName); // span2 - this shouldn't have changed, right?
print(noopScope2.span.operationName); // span2
```

I would expect the implementation to look more like:
```dart
/// The No-op implementation of [Scope] in which all operations are no-op
class NoopScope implements Scope {
  static final Span _noopSpan = new NoopSpan();

  /// Returns a new NoopScope, which will use the supplied Span. If no Span is
  /// supplied, a static instance of a NoopSpan will be used.
  NoopScope({Span span}) :
      _span = span ?? _noopSpan;

  final Span _span;

  @override
  void close() {}

  @override
  Span get span => _span;
}
```

Reply:
Good point